packages/core/src/lib/input/__tests__/utils.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { describe, it, expect } from 'vitest';
+import { getPatternForNumber } from '../utils';
+
+describe('Input utils', () => {
+  it.each([
+    { val: '1.2', allowNegative: true, expected: true },
+    { val: '-1.2', allowNegative: true, expected: true },
+    { val: '+1.2', allowNegative: true, expected: true },
+    { val: '1.2', allowNegative: false, expected: true },
+    { val: '-1.2', allowNegative: false, expected: false },
+    { val: '+1.2', allowNegative: false, expected: true },
+    { val: '0', allowNegative: true, expected: true },
+    { val: '-0', allowNegative: true, expected: true },
+    { val: '123', allowNegative: true, expected: true },
+    { val: '001.200', allowNegative: true, expected: true },
+    { val: '1.', allowNegative: true, expected: false },
+    { val: '.1', allowNegative: true, expected: false },
+    { val: 'abc', allowNegative: true, expected: false },
+    { val: '1.2.3', allowNegative: true, expected: false },
+    { val: '--1.2', allowNegative: true, expected: false },
+    { val: '1.2-', allowNegative: true, expected: false },
+    { val: '', allowNegative: true, expected: false },
+    { val: ' ', allowNegative: true, expected: false },
+  ])(
+    'validates floating-point numbers ($val)',
+    ({ val, allowNegative, expected }) => {
+      const pattern = getPatternForNumber('number', allowNegative);
+      expect(pattern.test(val)).toBe(expected);
+    }
+  );
```

Comment:
Tests are fairly overkill but easy enough to write & I think they are a good thing to have.